src/uct/api/uct.h

```diff
@@ -2564,6 +2568,13 @@ typedef struct {
      * "anonymous-uct_mem_alloc" is used by default.
      */
     const char                   *name;
+
+    /**
+     * Index of the system device on which memory is to be allocated.
+     * Eg: UCS_SYS_DEVICE_ID_UNKNOWN to allocate on host memory, or a specific
+     * index to allocate memory on GPU device.
```

Comment:
Just
"System device on which memory is to be allocated, or UCS_SYS_DEVICE_ID_UNKNOWN to allow allocating on any device".
The host memory comment is misleading because mem_type field is used to select host memory